frontend/src/APIClients/DonorAPIClient.ts

```diff
@@ -0,0 +1,63 @@
+import AUTHENTICATED_USER_KEY from "../constants/AuthConstants";
+import { DonorResponse } from "../types/DonorTypes";
+import { getLocalStorageObjProperty } from "../utils/LocalStorageUtils";
+import baseAPIClient from "./BaseAPIClient";
+
+const BEARER_TOKEN = `Bearer ${getLocalStorageObjProperty(
+  AUTHENTICATED_USER_KEY,
+  "accessToken",
+)}`;
+
+const getAllDonors = async (): Promise<DonorResponse[]> => {
+  try {
+    const { data } = await baseAPIClient.get("/donors", {
+      headers: { Authorization: BEARER_TOKEN },
+    });
+    return data;
+  } catch (error) {
+    return error as DonorResponse[];
+  }
+};
+
+const getDonorById = async (id: string): Promise<DonorResponse> => {
+  try {
+    const { data } = await baseAPIClient.get(`/donors/${id}`, {
+      headers: { Authorization: BEARER_TOKEN },
+    });
+    return data;
+  } catch (error) {
+    return error as DonorResponse;
+  }
+};
+
+const updateDonorById = async (
+  id: string,
+  donorData: any,
```

Comment:
Can you specify an `updateDonorData` type here instead of `any`? Look at `updateDonorDTO` in the backend folder `types.ts` file?

Reply:
it's what I meant in the earlier comment, we can't share types from server to client. the alternative is to type this as `FormData` or to add a new `UpdateUserDTO` type in the FE donor types file. Typing it as `any` is fine because of our validators and mapping that will only look at the relevant fields to begin with.

Or could consider exporting the types to a `libs` module or something for sharing

Reply:
yeah I meant adding in the FE donor types file